pkg/signup/service/signup_service.go

```diff
@@ -617,7 +585,7 @@ func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, username, phoneNumberOrHas
 		return errors.NewInternalError(err, "failed listing userSignups")
 	}
 	for _, signup := range userSignups {
-		if signup.Spec.Userid != userID && signup.Spec.Username != username && !states.Deactivated(signup) { // nolint:gosec
+		if signup.Spec.IdentityClaims.Sub != userID && signup.Spec.IdentityClaims.PreferredUsername != username && !states.Deactivated(signup) { // nolint:gosec
```

Comment:
Let's rename the `userID` param to something like `sub` in the function signature in the line 574, so it's bit less confusing.
```suggestion
		if signup.Spec.IdentityClaims.Sub != sub && signup.Spec.IdentityClaims.PreferredUsername != username && !states.Deactivated(signup) { // nolint:gosec
```